snuba/clickhouse/http.py

```diff
@@ -39,12 +44,34 @@ def __default(self, value: Any) -> Any:
         else:
             raise TypeError
 
-    def encode(self, value: WriterTableRow) -> JSONRow:
+    def encode(self, value: WriterTableRow) -> bytes:
         return cast(
             bytes, rapidjson.dumps(value, default=self.__default).encode("utf-8")
         )
 
 
+class ValuesRowEncoder(Encoder[bytes, WriterTableRow]):
+    def __init__(self, columns: Iterable[str]) -> None:
+        self.__columns = columns
+
+    def encode_value(self, value: Any) -> str:
+        if isinstance(value, ClickhouseUnguardedExpression):
+            return value.val
+        elif isinstance(value, ClickhouseInt):
+            return str(value.val)
+        elif isinstance(value, ClickhouseNumArray):
+            return "[" + ",".join([str(x) for x in value.val]) + "]"
```

Comment:
Two issues here:
- We are not escaping strings. If I received `asdasd'asdasd` This will produce an invalid statement for Clickhouse
- It is true that, on metrics we only have integers, but the encoder needs to work in a generic environment, we need to support floats and arrays of string and floats.

Also let's have a unit test of the encoder, with pytest parametrization you can write in a compact a way a large amount of test cases, which is a good scenario for the encoder.

Reply:
OK, though for this use case we do need to be able to push unescaped expressions to Clickhouse? Note also that I'm checking that at least the inputs to this method claim to be ClickhouseInt or ClickhouseNumArray, though as I think about it I guess python won't runtime enforce those types on the `val` property of those classes.

Are you thinking that this line:
```python
"percentiles": ClickhouseUnguardedExpression(
      f"arrayReduce('quantilesState(0.5,0.75,0.9,0.95,0.99)', {value_array})"
)
```

should be something like:
```python
# value_array is a ClickhouseNumArray
# encoding of ClickhouseStr would wrap with single quotes and ensure that single quotes inside are properly escaped
ClickhouseFunctionCall('arrayReduce', ClickhouseStr("quantilesState(0.5,0.75,0.9,0.95,0.99)"), value_array)
```

Reply:
> It is true that, on metrics we only have integers, but the encoder needs to work in a generic environment, we need to support floats and arrays of string and floats.

Is this actually true? I think all input should be handled properly in the sense of "throw an error for input we can't properly encode" but I don't see why we need to make this generalized if we only intend to use it for this one case (which itself is a bit of an experiment)

Reply:
> I think all input should be handled properly in the sense of "throw an error for input we can't properly encode"

A function that takes a string `f('asdasdasdasd')` would not fail to encode if the string needed escaping `asdasd'asdasd`. The insert in clickhouse would fail.
The problem, when that happens, is that the clickhouse INSERT would fail but since the INSERT is happening in batch, the batch itself will be too long to find the specific bad field (it gets truncated) and we won't know where the issue is. This already happened.

Anyway, if you want to implement only what is needed for this specific case for the sake of the experiment, that's ok, please add a TODO so you can provide a full implementation pf what is needed if the experiment works. 
Still please provide a unit test for the encoder.

Also we have a module with escaping function ready for use 
https://github.com/getsentry/snuba/blob/0429dc30390cc70221aa121b57057bf4ba324c98/snuba/clickhouse/escaping.py#L20-L42